### PR TITLE
Feature/strategy move

### PR DESCRIPTION
**Pull request by Demonsthere**
**Description**

Changes proposed in this pull request:

- Add accessStrategy field to CRD
- Disable validation tests, they need to be rewritten due to this change
- Update integration tests
- Mark `AuthStrategy` as depreciated
- Update samples
- Merge all processing into a single file, and single test file

**Related issue(s)**

- Implements #39

### CODE REVIEW

File: internal/processing/processing.go [hunk 1]
```diff
@@ -38,19 +35,122 @@ func NewFactory(vsClient *istioClient.VirtualService, arClient *oryClient.Access
...
+func (f *Factory) Run(ctx context.Context, api *gatewayv2alpha1.Gate) error {
+	var destinationHost string
+	var destinationPort uint32
+	var err error
+	var accessStrategies []*rulev1alpha1.Authenticator
+	// Get gate
+	// Get list of Paths
+	// Check Paths - validate
+	//Process path one by one
+	for i := range api.Spec.Rules {
```
Comment (Tomasz-Smelcerz-SAP) [inline internal/processing/processing.go]: It looks you don't need index at all. You can get elements directly: `for _, rule := range api.Spec.Rules {`

File: internal/processing/processing.go [hunk 2]
```diff
@@ -38,19 +35,122 @@ func NewFactory(vsClient *istioClient.VirtualService, arClient *oryClient.Access
...
+
+func (f *Factory) processAR(ctx context.Context, api *gatewayv2alpha1.Gate, config []*rulev1alpha1.Authenticator) error {
+	oldAR, err := f.getAccessRule(ctx, api)
+	if err != nil {
+		return err
+	}
+
+	if oldAR != nil {
+		ar := prepareAccessRule(api, oldAR, api.Spec.Rules[0], config)
+		// fmt.Printf("+++\n%+v\n", ar)
```
Comment (Tomasz-Smelcerz-SAP) [inline internal/processing/processing.go]: Remove

File: internal/processing/processing.go [hunk 3]
```diff
@@ -38,19 +35,122 @@ func NewFactory(vsClient *istioClient.VirtualService, arClient *oryClient.Access
...
+	var destinationPort uint32
+	var err error
+	var accessStrategies []*rulev1alpha1.Authenticator
+	// Get gate
+	// Get list of Paths
+	// Check Paths - validate
+	//Process path one by one
+	for i := range api.Spec.Rules {
+		// Single Path level
+		// Validate
```
Comment (Tomasz-Smelcerz-SAP) [inline internal/processing/processing.go]: Remove?

File: internal/processing/processing.go [hunk 4]
```diff
@@ -38,19 +35,112 @@ func NewFactory(vsClient *istioClient.VirtualService, arClient *oryClient.Access
...
+			destinationPort = *api.Spec.Service.Port
+		}
+		accessStrategies := []*rulev1alpha1.Authenticator{}
+		for i := range rule.AccessStrategy {
+			// Single Strategy for single Path
+			// Compile Oathkeeper config from this
+			accessStrategies = append(accessStrategies, rule.AccessStrategy[i])
+		}
+		// Create one AR per path
+		err = f.processAR(ctx, api, accessStrategies)
```
Comment (strekm) [inline internal/processing/processing.go]:
```suggestion
		err = f.processAR(ctx, api, rule.AccessStrategy)
```

File: internal/processing/processing.go [hunk 5]
```diff
@@ -38,19 +35,112 @@ func NewFactory(vsClient *istioClient.VirtualService, arClient *oryClient.Access
...
+	for _, rule := range api.Spec.Rules {
+		if isSecured(rule) {
+			destinationHost = f.oathkeeperSvc
+			destinationPort = f.oathkeeperSvcPort
+		} else {
+			destinationHost = fmt.Sprintf("%s.%s.svc.cluster.local", *api.Spec.Service.Name, api.ObjectMeta.Namespace)
+			destinationPort = *api.Spec.Service.Port
+		}
+		accessStrategies := []*rulev1alpha1.Authenticator{}
+		for i := range rule.AccessStrategy {
```
Comment (strekm) [inline internal/processing/processing.go]: this block of code is not needed

**Pull request merged**